src/harness.ts

```diff
@@ -35,34 +36,51 @@ let harnessId = 0;
  * @param target the root DNode to search
  * @param key the key to match
  */
-function findDNodeByKey(target: DNode, key: string | object): HNode | WNode | undefined {
-	if (target && typeof target === 'object') {
-		if (target && typeof target === 'object' && target.properties && target.properties.key === key) {
-			return target;
-		}
+function findDNodeByKey(target: RenderResults, key: string | object): HNode | WNode | undefined {
+	if (!target) {
+		return;
+	}
+	if (target instanceof Array) {
 		let found: HNode | WNode | undefined;
-		if (target.children) {
-			target.children
-				.forEach((child) => {
-					if (found) {
-						if (findDNodeByKey(child, key)) {
-							console.warn(`Duplicate key of "${key}" found.`);
-						}
-					}
-					else {
-						found = findDNodeByKey(child, key);
-					}
-				});
-		}
+		target.forEach((node) => {
+			if (found) {
+				if (findDNodeByKey(node, key)) {
+					console.warn(`Duplicate key of "${key}" found.`);
+				}
+			}
+			else {
+				found = findDNodeByKey(node, key);
+			}
+		});
 		return found;
+	} else {
```

Comment:
current style guide is:

```ts
}
else {
```

Reply:
Ugh, why?

Reply:
It is a Colin hold over, which actually makes it significantly easier when refactoring code, because you can just cut out blocks (or insert blocks) without having to rematch braces.  For example:

```diff
if (true) {
  console.log('true');
}
-else {
-  console.log('false');
-}
```

Instead of:

```diff
if (true) {
  console.log('true');
+}
-} else {
-  console.log('false');
-}
```